pkg/filesystem/local_directory_unix.go

```diff
@@ -33,8 +33,17 @@ func newLocalDirectoryFromFileDescriptor(fd int) (*localDirectory, error) {
 
 // NewLocalDirectory creates a directory handle that corresponds to a
 // local path on the system.
-func NewLocalDirectory(path string) (DirectoryCloser, error) {
-	fd, err := unix.Openat(unix.AT_FDCWD, path, unix.O_DIRECTORY|unix.O_NOFOLLOW|unix.O_RDONLY, 0)
+func NewLocalDirectory(directoryParser path.Parser) (DirectoryCloser, error) {
+	directoryPath, scopeWalker := path.EmptyBuilder.Join(path.NewAbsoluteScopeWalker(path.VoidComponentWalker))
```

Comment:
Maybe better to use VoidScopeWalker here. Right now we don't require that this function takes an absolute path.

Reply:
:+1: